src/lib/models/ConfigModel.ts

```diff
@@ -10,48 +11,85 @@ export interface ConfigModel {
-config_root: null # Last calculated by the config command as: <COMING_SOON>
-
-# Folder where to store log files. May be an absolute path or a path relative to config_root.
-# If not specified or if equal to null, then the value is automatically chosen to be your machine's default temporary directory
-log_folder: null  # Last calculated by the config command as: <COMING_SOON>
-
-# (Use at your own risk. This is an experimental property that is not officially supported.)
-# List of EnginePlugin module paths to dynamically add to Code Analyzer. Each path may be absolute or relative to config_root.
-custom-engine_plugin_modules: []
-
-# Rule override settings of the format rules.<engine_name>.<rule_name>.<property_name> = <override_value> where
-#   * <engine_name> is the name of the engine containing the rule that you want to override
-#   * <rule_name> is the name of the rule that you want to override
-#   * <property_name> can either be:
-#      'severity' - [Optional] The severity level value that you want to use to override the default severity level for the rule
-#                   Possible values: 1 or 'Critical', 2 or 'High', 3 or 'Moderate', 4 or 'Low', 5 or 'Info'
-#      'tags'     - [Optional] The string array of tag values that you want to use to override the default tags for the rule
-#
-# [Example usage]:
-# rules:
-#   eslint:
-#     sort-vars:
-#       severity: Info
-#       tags: ["Recommended", "Suggestion"]
-rules: {} # Remove this empty object {} when you are ready to specify your first rule override
-
-# Engine specific custom configuration settings of the format engines.<engine_name>.<property_name> = <value> where
+function yamlFormatConfig(rawConfig: CodeAnalyzerConfig, ruleSelection: RuleSelection): string {
+
+	let results = yamlFormatConfigProperty('config_root', rawConfig.getConfigRoot(), true)
+		+ '\n'
+		+ yamlFormatConfigProperty('log_folder', rawConfig.getLogFolder(), true)
+		+ '\n';
+	const customEngineModules: string[] = rawConfig.getCustomEnginePluginModules();
+	const customEngineModulesString: string = customEngineModules.length > 0
+		? `["${customEngineModules.join('", "')}"]`
+		: `[]`;
+	results += yamlFormatConfigProperty('custom_engine_plugin_modules', customEngineModulesString, false)
+		+ '\n'
+		+ yamlFormatRuleSelection(ruleSelection)
+		+ '\n'
+		+ DUMMY_ENGINE_SECTION
+		+ '\n';
+	return results;
+}
+
+function yamlFormatConfigProperty(propertyName: string, propertyValue: string, isCalculated: boolean): string {
+	// Get the annotation and prepend each line with '# ' to make it a YAML comment.
+	const annotation: string = getMessage(BundleName.ConfigModel, `annotation.${propertyName}`)
+		.replace(/^./gm, s => `# ${s}`);
+	const lastCalculatedAs: string = getMessage(BundleName.ConfigModel, 'template.last-calculated-as', [propertyValue]);
+
+	return `${annotation}\n`
+		+ `${propertyName}: ${propertyValue}${isCalculated ? ' # ' + lastCalculatedAs : ''}\n`;
+}
```

Comment:
I'm a little confused here. This appears to show the propertyValue being set and added to the comment.

The original idea behind the comment was for values that the user has not explicitly set that get calculated. For example, if the user has not set the log_folder... then it should remain null but give a hint as to what the calculated value is. For example:

```
log_folder: null  # Last calculated by the config command to be: '/var/folders/1x/9xkd2bj500z58k2s5b8f2y980000gp/T/`
```

We do not want to override the null value with a non-null value unless the user explicitly did it themselves because the default value isn't fixed here.

To detect if a user actually supplied a non-default value themselves or not... we can just construct a CodeAnalyzerConfig.withDefaults and compare their values against the default values. If their value matches the default value... then we just put null for these path based derived fields like config_root, log_folder, etc. with a comment saying what the derived value was.

This is important that we keep null because if they check this into a CI/CD system... they most likely want to not have paths that are absolute to their local machine.

We can discuss more during tea time if you want.

Reply:
Okay, then it sounds like we miscommunicated about the expected behavior, because my understanding was that the `last calculated as blah` comment was to allow you to change the values that the command derived without losing whatever the original value was.
If we want to preserve null values and have the comment indicate the derived value, then yeah we can do that.

Reply:
Yeah... specifically for the derived defaults which depend on the system environment... best to keep those as null - so that if the system changes (like in the CI/CD pipelines for example) then we still can do the lookup to derive the value based on the environment.